### ISSUE TITLE
Mass lightmap export and support for Portal 2 and CS GO branches

### ISSUE THREAD

**Issue by luxeleios** [Sergey-KoRJiK/SourceEngineBSPditor#1]
Hey, first of all, thank you for your efforts. I know this release is a few years old, but I was wondering if its possible for you to implement mass/batch lightmap loading and saving, so that you don't have to do it one by one, and if you can implement support for later source branches - Portal 2 and CS GO - the main difference is that Portal 2 doesn't use LDR lighting and it saves lightmaps to a different lump, I believe. I can get you more specific info if you wish. In any case, thank you for your work.

**Comment by Sergey-KoRJiK** [comment]
> Hey, first of all, thank you for your efforts. I know this release is a few years old, but I was wondering if its possible for you to implement mass/batch lightmap loading and saving, so that you don't have to do it one by one, and if you can implement support for later source branches - Portal 2 and CS GO - the main difference is that Portal 2 doesn't use LDR lighting and it saves lightmaps to a different lump, I believe. I can get you more specific info if you wish. In any case, thank you for your work.

Generally speaking, I'm already working on the idea of ​​the 3rd option, but inside my editor: this is the selection of several faces (with additional options of fast group selection by vis-cluster, both material/brush and ect.) in the 3d window and then creating a common UV with a separate 2d window (with options to create a layers of individual selection groups for fast switching when u work with few areas on map). That is, before your eyes you will have a 3D view of selected faces with lightmaps, and a 2D scan window of these lightmaps. At the same time, the 2D window will have the functionality of copying/pasting from/to the buffer, as well as the basic functionality of spreading/filling/blurring with brushes similar to PhotoShop with masking (and some color correction and tonemapping functionality for HDR images). I see this approach as the most optimal and simple solution. And any change in 2D window, by possibility, update in realtime in 3D window.

**Comment by luxeleios** [comment]
Wait, do you speak russian? I actually didn't realize that up until now. Let me know if you're willing to continue the discussion on discord. 
The reason I requested mass loading and saving is indeed because I wanted to try to apply different photoshop filters to lightmaps and see what I can get out of it, so for that purpose it would be useful to export to texture atlases (you could even export multiple high res ones), I think. 
However the obj + UV method seems like the most optimal for everything else, and I actually have a few ideas I'd like to discuss, and I'm not sure if it's best to do it here.

**Comment by Sergey-KoRJiK** [comment]
> Wait, do you speak russian? I actually didn't realize that up until now. Let me know if you're willing to continue the discussion on discord. The reason I requested mass loading and saving is indeed because I wanted to try to apply different photoshop filters to lightmaps and see what I can get out of it, so for that purpose it would be useful to export to texture atlases (you could even export multiple high res ones), I think. However the obj + UV method seems like the most optimal for everything else, and I actually have a few ideas I'd like to discuss, and I'm not sure if it's best to do it here.

Yeah, we can continue discussion in discord. My discord id in account info (Sergey#5205)
